doc/api/esm.md

```diff
@@ -44,822 +44,41 @@ Node.js contains support for ES Modules based upon the
-
-import 'commonjs-package';
-// Loaded as CommonJS since ./node_modules/commonjs-package/package.json
-// lacks a "type" field or contains "type": "commonjs".
-
-import './node_modules/commonjs-package/index.js';
-// Loaded as CommonJS since ./node_modules/commonjs-package/package.json
-// lacks a "type" field or contains "type": "commonjs".
-```
-
-Files ending with `.mjs` are always loaded as ES modules regardless of package
-scope.
-
-Files ending with `.cjs` are always loaded as CommonJS regardless of package
-scope.
-
-```js
-import './legacy-file.cjs';
-// Loaded as CommonJS since .cjs is always loaded as CommonJS.
-
-import 'commonjs-package/src/index.mjs';
-// Loaded as ES module since .mjs is always loaded as ES module.
-```
-
-The `.mjs` and `.cjs` extensions may be used to mix types within the same
-package scope:
-
-* Within a `"type": "module"` package scope, Node.js can be instructed to
-  interpret a particular file as CommonJS by naming it with a `.cjs` extension
-  (since both `.js` and `.mjs` files are treated as ES modules within a
-  `"module"` package scope).
-
-* Within a `"type": "commonjs"` package scope, Node.js can be instructed to
-  interpret a particular file as an ES module by naming it with an `.mjs`
-  extension (since both `.js` and `.cjs` files are treated as CommonJS within a
-  `"commonjs"` package scope).
-
-### `--input-type` flag
-
-Strings passed in as an argument to `--eval` (or `-e`), or piped to `node` via
-`STDIN`, will be treated as ES modules when the `--input-type=module` flag is
-set.
-
-```bash
-node --input-type=module --eval "import { sep } from 'path'; console.log(sep);"
-
-echo "import { sep } from 'path'; console.log(sep);" | node --input-type=module
-```
-
-For completeness there is also `--input-type=commonjs`, for explicitly running
-string input as CommonJS. This is the default behavior if `--input-type` is
-unspecified.
+Node.js treats JavaScript code as CommonJS modules by default.
+Authors can tell Node.js to treat JavaScript code as ECMAScript modules
+via the `.mjs` file extension, the `package.json` `"type"` field, or the
+`--input-type` flag. See
+[Modules: Packages](packages.html#packages_determining_module_system) for more
+details.
+
+<!-- Anchors to make sure old links find a target -->
+<i id="esm_package_entry_points"></i>
+<i id="esm_main_entry_point_export"></i>
+<i id="esm_subpath_exports"></i>
+<i id="esm_package_exports_fallbacks"></i>
+<i id="esm_exports_sugar"></i>
+<i id="esm_conditional_exports"></i>
+<i id="esm_nested_conditions"></i>
+<i id="esm_self_referencing_a_package_using_its_name"></i>
+<i id="esm_internal_package_imports"></i>
+<i id="esm_dual_commonjs_es_module_packages"></i>
+<i id="esm_dual_package_hazard"></i>
+<i id="esm_writing_dual_packages_while_avoiding_or_minimizing_hazards"></i>
+<i id="esm_approach_1_use_an_es_module_wrapper"></i>
+<i id="esm_approach_2_isolate_state"></i>
```

Comment:
Is there a reason for not just making these `<a name="">` tags?

Reply:
That's was my original approach, however:

* `name` attribute is deprecated for `<a>` elements in HTML5.
* `<a>` elements get decorated with CSS – even when they are empty. You can actually observe that in https://nodejs.org/docs/latest/api/deprecations.html, when you hover above the `DEPXXXX` heading:
    - ![image](https://user-images.githubusercontent.com/14309773/91336880-e3401e80-e7d2-11ea-8a1f-da65fd3e05b4.png)
    - ![image](https://user-images.githubusercontent.com/14309773/91336924-f18e3a80-e7d2-11ea-894e-9af9254169ad.png)

If `<i>` is not appropriate, I'm fine using something else BTW.

Reply:
Ok was just curious, I'm fine with this approach. I did not know the `"name"` element was deprecated! So `window.esm_conditional_exports` being defined it is :P